lms/djangoapps/instructor/views/instructor_dashboard.py

```diff
@@ -488,7 +494,15 @@ def _section_membership(course, access):
             'update_forum_role_membership',
             kwargs={'course_id': str(course_key)}
         ),
-        'is_reason_field_enabled': configuration_helpers.get_value('ENABLE_MANUAL_ENROLLMENT_REASON_FIELD', False)
+        'is_reason_field_enabled': configuration_helpers.get_value('ENABLE_MANUAL_ENROLLMENT_REASON_FIELD', False),
+
+        # Membership section should be hidden for eSHE instructors.
+        # Since they get Course Staff role implicitly, we need to hide this
+        # section if the user doesn't have the Course Staff role set explicitly
+        # or have the Discussion Admin role.
+        'is_hidden': (
+            access['eshe_instructor'] and not (access['explicit_staff'] or access['forum_admin'])
```

Comment:
@0x29a I would regroup it like:
```suggestion
            not access['forum_admin'] and (access['eshe_instructor'] and not access['explicit_staff'])
```
because semantically it makes more sense, e.g. we always display it if user is `forum_admin`, and if they are an `eshe_instructor` we have to check if they are `explicit_staff` or not.

Reply:
Actually, should this section be visible if a user is a `staff`, but not a `forum_admin`?

Reply:
@Cup0fCoffee, yes, should be visible for `staff`, as they by default it has access to bulk enrollments / assigning course team roles.